### PR TITLE
GDB-7337: Implements click on main menu plugin

### PR DESCRIPTION
**Pull request by boyan-tonchev**
WHAT:

     Added 'click-main-menu' guide plugin.
WHY:

	It can used as part of other guide plugin when have interaction with main menu.
HOW:

	Registers 'click-main-menu' plugin to 'guide.step' plugins.

Additional changes:

   Extends guide service with new functions.
   Added specific guide selector to all menus and sub menus.
   Added html support to popover dialog. Translate service is configured to escape HTML entities as <,> etc. and this breaks processing dialog content as html. Fixed as unescapes escaped characters of translated content.

### CODE REVIEW

File: src/js/angular/guides/steps/complex/main-menu/plugin.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+                steps.push({
+                    'guideBlockName': 'clickable-element',
+                    'options': angular.extend({}, {
+                        'title': 'guide.step_plugin.click-main-menu.' + options.label + '.main-menu.title',
+                        'content': 'guide.step_plugin.click-main-menu.' + options.label + '.main-menu.content',
+                        'elementSelector': GuideUtils.getGuideElementSelector(options.parentMenuSelector),
+                        // Open main menu if is not open
+                        'onNextClick': () => {
+                            // first we check if sub menu is not visible this mean that parent menu is not open.
+                            if (!angular.element(document.querySelector(GuideUtils.getGuideElementSelector(options.menuSelector))).is(':visible')) {
```
Comment (teodossidossev) [inline src/js/angular/guides/steps/complex/main-menu/plugin.js:17]: and existing check maybe

File: src/js/angular/guides/steps/complex/main-menu/plugin.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+                    'title': 'guide.step_plugin.click-main-menu.' + options.label + '.menu.title',
+                    'content': 'guide.step_plugin.click-main-menu.' + options.label + '.menu.content',
+                    'elementSelector': GuideUtils.getGuideElementSelector(options.menuSelector),
+                    'placement': 'left',
+                    'onNextClick': () => {
+                        GuideUtils.clickOnGuideElement(options.menuSelector, ' a')()
+                    },
+                    'onPreviousClick': () => {
+                        // first we check if sub menu is not visible this mean that parent menu is not open.
+                        if (!angular.element(document.querySelector(GuideUtils.getGuideElementSelector(options.menuSelector))).is(':visible')) {
```
Comment (teodossidossev) [inline src/js/angular/guides/steps/complex/main-menu/plugin.js:45]: same here

**Comment by teodossidossev** [comment]
lgtm

File: src/js/angular/guides/steps/complex/main-menu/plugin.js (hunk 1 above)
Comment (boyan-tonchev) [reply src/js/angular/guides/steps/complex/main-menu/plugin.js]: done

File: src/js/angular/guides/steps/complex/main-menu/plugin.js (hunk 2 above)
Comment (boyan-tonchev) [reply src/js/angular/guides/steps/complex/main-menu/plugin.js]: done

**Review by teodossidossev — APPROVED** [review]

**Comment by boyan-tonchev** [comment]
Jenkins retry build

**Pull request merged**